usefulUtilities/statusIndicator/hifiScript/statusIndicator.js

```diff
@@ -0,0 +1,210 @@
+        queryParamString += "&displayName=" + MyAvatar.displayName;
+        queryParamString += "&status=";
+        queryParamString += isAvailable ? "available" : "busy";
+
+        if (DEBUG) {
+            console.log("sendStatusUpdate: " + REQUEST_URL + queryParamString);
+        }
+
+        request({
+            uri: REQUEST_URL + "?" + queryParamString
+        }, function (error, response) {
+            if (error || !response || response.status !== "success") {
+                console.error("Error with sendStatusUpdate: " + JSON.stringify(response));
+                return;
+            }
+        });
+    }
+
+    // #endregion SEND STATUS REQUEST
+
+
+    // #region SIGNALS
+
+    // On mouse press, check if the status overlay on desktop was clicked
+    // If yes, change status to the opposite and send status update
+    var isAvailable = true; // Default is available
+    function onMousePressEvent(event) {
+        if (DEBUG) {
+            console.log("onMousePressEvent isLeftButton: " + event.isLeftButton);
+        }
+        // is primary button
+        var overlayID = Overlays.getOverlayAtPoint({ x: event.x, y: event.y });
+        if (event.isLeftButton && overlayID && (overlayID === rectangleOverlay || overlayID === desktopOverlay)) {
+            if (DEBUG) {
+                console.log("onMousePressEvent status overlay has been clicked");
+            }
+            isAvailable = !isAvailable;
+            var edits = {};
+            edits[rectangleOverlay] = { color: isAvailable ? AVAILABLE_COLOR : NOT_AVAILABLE_COLOR }
+            edits[desktopOverlay] = { text: isAvailable ? OVERLAY_AVAILABLE_TEXT : OVERLAY_NOT_AVAILABLE_TEXT }
+            Overlays.editOverlays(edits);
+
+            sendStatusUpdate();
+        }
+    }
+
+
+    // When window resizes, redraw overlays
+    function onWindowResize() {
+        deleteStatusOverlays();
+        drawStatusOverlays();
+    }
+
+    
+    // When avatar becomes active from being away
+    // Set status back to previousStatus
+    function onWentActive() {
+        isAvailable = previousStatus;
+        onWindowResize();
+        sendStatusUpdate();
+    }
+
+
+    // When avatar goes away, set status to busy
+    var previousStatus;
+    function onWentAway() {
+        previousStatus = isAvailable;
+        isAvailable = false;
+        onWindowResize();
+        sendStatusUpdate();
+    }
+
+    // #endregion SIGNALS
+
+
+    // #region APP LIFETIME
+
+    // Creates the app button and sets up signals and hearbeat
+    function startup() {
+        drawStatusOverlays();
+
+        Script.scriptEnding.connect(unload);
+        Controller.mousePressEvent.connect(onMousePressEvent);
+        Window.geometryChanged.connect(onWindowResize);
+        MyAvatar.wentAway.connect(onWentAway);
+        MyAvatar.wentActive.connect(onWentActive);
+        MyAvatar.displayNameChanged.connect(sendStatusUpdate);
+
+        setupHeartbeat();
```

Comment:
you're going to want to call `sendStatusUpdate` just once right after this, right? otherwise the server will have to wait N seconds before receiving the first status update

Reply:
agreed